src/bauhaus/bauhaus.c

```diff
@@ -1552,6 +1555,16 @@ gboolean dt_bauhaus_combobox_set_from_value(GtkWidget *widget, int value)
       return TRUE;
     }
   }
+
+  dt_introspection_type_enum_tuple_t *values
```

Comment:
But I don't get this fallback... If the item is not found in the loop above we come here...

Maybe some comments would help?

Reply:
This is really unrelated to the original bugreport (the solution of which is just the part above in dt_bauhaus_combobox_add_introspection).

Here, whenever an enum value is not currently in the combo (presumably because it is legacy and was manually removed in gui_init after the combo was created _from_param because we don't want _new_ images to use it), if its value is being selected (presumably from loading an old image that used the legacy value) we try to add it back to the combo from introspection . This means we don't have to do it manually in gui_update, which is messy. So filmic and highlights (and denoiseprofile) can be simplified.